pkg/util/kubernetes/hpa/hpa.go

```diff
@@ -119,11 +119,11 @@ func newHPAWatcher() *HPAWatcherClient {
 		refreshItl:     time.NewTicker(time.Duration(hpaRefreshItl) * time.Second),
 		externalMaxAge: time.Duration(hpaExternalMaxAge) * time.Second,
 		ns:             namespace,
-	}
+	}, nil
 }
 
 // GetHPAWatcherClient returns the HPAWatcherClient
-func GetHPAWatcherClient() *HPAWatcherClient {
+func GetHPAWatcherClient() (*HPAWatcherClient, error) {
```

Comment:
I think you could just make `NewHPAWatcherClient` public instead of having this function.